### ISSUE TITLE
Review_LP_20170927

### ISSUE THREAD

**Issue by lukpueh** [aaaaalbert/fogwc#4]
Below my review (disguised as an issue). I actually don't have much to criticize, just two minor things:

__Roles__
https://github.com/aaaaalbert/fogwc/blob/a85cd0fdc5415c5a54b2135435431b83d03c54e4/architecture.tex#L89-L91

I am not sure if experimenters' privileges are already clear enough to justify the conclusion: *experimenters have keys ergo can authenticate each other and swap resources*. 
Maybe we should very briefly explain the possible capacities of the role and what *sandbox control* means?

Device/Sandbox privileges are actually explained in the newly added section about *Trust Management* but we use different terminology there, i.e. *Admin* and *User*.
I suspect @aaaaalbert of having used these terms because they are better suited to talk about privileges, whereas the terms *device owners* and *experimenters* are are more ambiguous in this regard.?
In any case I think we need some sort of mapping between those 4 roles.

__Uniform case and italicization for roles and components__
We still have some case and text emphasis inconsistency for roles (device owners, experimenters, operators, Admin, User) and components (Sandbox, Resource Manager, Device Manager, Experiment Manager, Installer Builder, NAT Relays, Clearinghouse).

What's the best way to do this? First mention italic, rest normal, always capitalized?

**Comment by jhdalek55** [comment]
I'm not going to chime in on the first issue you raise, and for the second, while I don't know whether device owner and experimenter is better than admin and user, I do agree that we need to consistently use the same terms. Admin and user are certainly shorter, and I agree that they might be less ambiguous. But, pick one set and stick with it.

As for case and italicization, I think capitalizing things that are not really proper nouns gets a bit annoying. I like Lukas' suggestion of italicizing the term when it is first introduced, and then leaving lower case and unitalicized for the rest of the paper.

I'm going to review the paper tomorrow morning. Thanks.

Lois

**Comment by aaaaalbert** [comment]
Thanks for your feedback!

@lukpueh:
> I am not sure if experimenters' privileges are already clear enough to justify the conclusion: experimenters have keys ergo can authenticate each other and swap resources.
Maybe we should very briefly explain the possible capacities of the role and what sandbox control means?

This is an artifact of the "Trust" subsection being bolted on without big modifications to the other subsections. I think we can streamline this quite easily.

> Uniform case and italicization for roles and components

I attempted to format stuff consistently, but you be the judges. I *Capitalize And Italicize* a term the first time I use it; afterwards it's Capitalized if referring to the category (like "Experiment Managers control Sandboxes", generally), and lower-case when talking about instances (not in the paper, but among the lines of "Bob asks Alice to use his clearinghouse"). However, we can probably drop the capitalization.

@jhdalek55:
> (...) while I don't know whether device owner and experimenter is better than admin and user, I do agree that we need to consistently use the same terms (...)

These are actually four potentially-indepent roles. Admin and User (capitalized) is what the technical implementation of the sandbox understands. Device owners and experimenters are real people who might or might not act as Admins and Users elsewhere.

**Comment by aaaaalbert** [comment]
Since @justincappos also found the capitalization confusing, I'll remove it.

**Comment by aaaaalbert** [comment]
I also substituted fog/Fog and cloud/Cloud, so they are all lower case now.

**Comment by jhdalek55** [comment]
I'm borrowing Lukas' "issue," rather than opening a new one.

I just did a full pass on the paper and have submitted a pull request from github desktop branch. Mostly these were just sentence level corrections or wording changes for clarity, but I did have a few larger level questions/comments.

The first is the use of the word "introduced" as related to Seattle. It's used in both the introduction and conclusion. It sounds a little odd to use the word "introduce" for a testbed that you describe as being in use for 8 years. "Introduce" generally means this is something we haven't seen before. From my perspective, this paper is proposing the use of this existing testbed called Seattle in a new and exciting way to meet the demands of this emerging technology. No, I'm not proposing those as the words to use, because that is way too wordy (maybe instead it could just read "In this paper, we make the case for why our Seattle platform is uniquely designed for fog computing deployments." 

Second, contribution three in the introduction does not sound like we are doing something new...hence, is it it a contribution?

And, third, in Section IIA., do we need to explain that the sandbox is downloaded to the user phone?

That's it. It's 3 p.m and I need lunch.

**Comment by aaaaalbert** [comment]
Thanks!

* As mentioned on #5, I've merged virtually all of your edits.
* We now "present" Seattle in both the Intro and the Conclusion, e9302b28ef745a1742d8fc0f45b0701cc546e44e.
* I think our live deployment is a contribution in the space of fog computing. Most other papers describe the bright future that could be achieve if only everybody switched to smart traffic lights and whatnot. We on the other hand run our system on actual devices "out there".
* The download aspect is there, but it's not up front. It is mentioned half a column further down where we talk about the installer builder.

**Comment by jhdalek55** [comment]
The only argument I might make is about the third contribution..It could be
tweaked to read a bit more like a contribution with the addition of a few
words

" We validate Seattle's ability to facilitate fog computing systems by
presenting our live deployment. Already installed on tens of thousands of
devices, it has been used by...."

Might be a bit wordy, but I think  it better connects Seattle's track
record to the purpose of the paper. But then again, time is growing short
and I know you want to finish this, so I leave it to you.

Lois

**Comment by lukpueh** [comment]
Thanks @jhdalek55, I adopted your suggestion in f633cb1524ca05a5a4778726cf13e3dc20249b8b

**Comment by aaaaalbert** [comment]
:+1:

**Comment by jhdalek55** [comment]
Cool. Will now delete branch.

Lois
